plugins/lookup-table/src/main/java/pl/edu/uj/kimage/plugins/lookup/table/InvertFlowStep.java

```diff
@@ -0,0 +1,55 @@
+package pl.edu.uj.kimage.plugins.lookup.table;
+
+import pl.edu.uj.kimage.api.Step;
+import pl.edu.uj.kimage.api.StepDependency;
+import pl.edu.uj.kimage.eventbus.EventBus;
+import pl.edu.uj.kimage.plugin.FlowStep;
+import pl.edu.uj.kimage.plugin.ImageLoaded;
+import pl.edu.uj.kimage.plugin.StepResultEvent;
+import pl.edu.uj.kimage.plugin.model.Color;
+import pl.edu.uj.kimage.plugin.model.Image;
+import pl.edu.uj.kimage.plugin.model.ImageBuilder;
+
+import java.util.List;
+
+public class InvertFlowStep extends FlowStep {
+
+    public InvertFlowStep(Step step, EventBus eventBus) {
+        super(step, eventBus);
+    }
+
+    @Override
+    public void processRelatedEvent(StepResultEvent event) {
+        List<StepDependency> dependencies = getStep().getDependencies();
+        StepDependency stepDependency = dependencies.get(0);
+        String eventTypeName = stepDependency.getEventTypeName();
+
+        try {
+            Class<?> clazz = Class.forName(eventTypeName);
+            if (clazz.isInstance(ImageLoaded.class)) {
+
+                ImageLoaded imageLoaded = (ImageLoaded) event;
+                Image input = imageLoaded.getLoadedImage();
+
+                int height = input.getHeight();
+                int width = input.getWidth();
+
+                ImageBuilder builder = new ImageBuilder(width, height);
+                for (int yCounter = 0; yCounter < height; yCounter++) {
+                    for (int xCounter = 0; xCounter < width; xCounter++) {
+                        Color color = input.getColor(xCounter, yCounter);
+                        Color invertedColor = Color.WHITE.minus(color.getRed(), color.getGreen(), color.getBlue(), color
+                                .getAlpha());
+                        builder.withColor(invertedColor, xCounter, yCounter);
+                    }
+                }
+
+                Image invertedImage = builder.build();
+                StepResultEvent result = new ImageLoaded(event.getStepNumber(), invertedImage);
```

Comment:
Why not?

On Thu, May 19, 2016 at 11:19 PM, annadobrowolska < notifications@github.com [notifications@github.com] > wrote:
In plugins/lookup-table/src/main/java/pl/edu/uj/kimage/plugins/lookup/table/InvertFlowStep.java [https://github.com/kmisztal/KiMage/pull/48#discussion_r63953204] :

> +
> +                int height = input.getHeight();
> +                int width = input.getWidth();
> +
> +                ImageBuilder builder = new ImageBuilder(width, height);
> +                for (int yCounter = 0; yCounter < height; yCounter++) {
> +                    for (int xCounter = 0; xCounter < width; xCounter++) {
> +                        Color color = input.getColor(xCounter, yCounter);
> +                        Color invertedColor = Color.WHITE.minus(color.getRed(), color.getGreen(), color.getBlue(), color
> +                                .getAlpha());
> +                        builder.withColor(invertedColor, xCounter, yCounter);
> +                    }
> +                }
> +
> +                Image invertedImage = builder.build();
> +                StepResultEvent result = new ImageLoaded(event.getStepNumber(), invertedImage);


ImageLoaded is common class in pl.edu.uj.kimage.plugin.model package, do you want to create separate event with the same content for each plugin?

—
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub [https://github.com/kmisztal/KiMage/pull/48/files/dfc59047dcfb11709bd59fd068553783409c6e2e#r63953204]